eclair-core/src/main/scala/fr/acinq/eclair/db/migration/CompareDb.scala

```diff
@@ -0,0 +1,80 @@
+    }
+
+    res
+  }
+
+  // @formatter:off
+  import fr.acinq.eclair.db.jdbc.JdbcUtils.ExtendedResultSet._
+  def bytes(rs: ResultSet, columnName: String): ByteVector = rs.getByteVector(columnName)
+  def bytesnull(rs: ResultSet, columnName: String): ByteVector = rs.getByteVectorNullable(columnName).getOrElse(ByteVector.fromValidHex("deadbeef"))
+  def hex(rs: ResultSet, columnName: String): ByteVector = rs.getByteVectorFromHex(columnName)
+  def hexnull(rs: ResultSet, columnName: String): ByteVector = rs.getByteVector32FromHexNullable(columnName).map(_.bytes).getOrElse(ByteVector.fromValidHex("deadbeef"))
+  def string(rs: ResultSet, columnName: String): ByteVector = ByteVector(rs.getString(columnName).getBytes)
+  def stringnull(rs: ResultSet, columnName: String): ByteVector = ByteVector(rs.getStringNullable(columnName).getOrElse("<null>").getBytes)
+  def bool(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromByte(if (rs.getBoolean(columnName)) 1 else 0)
+  def long(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getLong(columnName))
+  def longnull(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getLongNullable(columnName).getOrElse(42))
+  def longts(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong((rs.getLong(columnName).toDouble / 1_000_000).round)
+  def longtsnull(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getLongNullable(columnName).map(l => (l.toDouble/1_000_000).round).getOrElse(42))
+  def int(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromInt(rs.getInt(columnName))
+  def ts(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong((rs.getTimestamp(columnName).getTime.toDouble / 1_000_000).round)
+  def tsnull(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getTimestampNullable(columnName).map(t => (t.getTime.toDouble / 1_000_000).round).getOrElse(42))
+  def tssec(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong((rs.getTimestamp(columnName).toInstant.getEpochSecond.toDouble / 1_000_000).round)
+  def tssecnull(rs: ResultSet, columnName: String): ByteVector = ByteVector.fromLong(rs.getTimestampNullable(columnName).map(t => (t.toInstant.getEpochSecond.toDouble / 1_000_000).round).getOrElse(42))
+  // @formatter:on
+
+  def compareAll(dualDatabases: DualDatabases): Unit = {
+    logger.info("comparing all tables...")
+    val (sqliteDb, postgresDb) = DualDatabases.getDatabases(dualDatabases)
+    PgUtils.inTransaction { postgres =>
+      val result = List(
+        CompareChannelsDb.compareAllTables(sqliteDb.channels.sqlite, postgres),
+        ComparePendingCommandsDb.compareAllTables(sqliteDb.channels.sqlite, postgres),
+        ComparePeersDb.compareAllTables(sqliteDb.channels.sqlite, postgres),
+        ComparePaymentsDb.compareAllTables(sqliteDb.channels.sqlite, postgres),
```

Comment:
I know these are all using the same connection under the hood, but it's weird to read them all using `sqlite.channels` when the two below use something different. I think it's better to make them explicitly correct:

```suggestion
        CompareChannelsDb.compareAllTables(sqliteDb.channels.sqlite, postgres),
        ComparePendingCommandsDb.compareAllTables(sqliteDb.pendingCommands.sqlite, postgres),
        ComparePeersDb.compareAllTables(sqliteDb.peers.sqlite, postgres),
        ComparePaymentsDb.compareAllTables(sqliteDb.payments.sqlite, postgres),
```

Reply:
Done with 5c3eac546.